File: database-go/www/handlers/clear.go
package handlers

import (
	"github.com/labstack/echo/v4"
	"go-kv-database/app"
	"go-kv-database/storage"
	"net/http"
)

func Clear(c echo.Context) error {
	// Database actions
	storage.Clear()

	// Request replicate to the other nodes
	app.AsyncReplicateMeshRequest(c.Path(), c.QueryParams(), true)

	// Response
	return c.JSON(http.StatusOK, map[string]bool{
		"success": true,
	})
}

File: packages/database/src/http/handlers/clear.ts
import { RequestHandler } from '../../net/http';

// . Clear all records [mutable]
export const clear: RequestHandler = async (db) => {
  // Database actions [no waiting to execute]
  db.clear()
    .catch(console.error);

  // Response
  return {
    success: true
  };

  /*
  // @todo
	// Request replicate to the other nodes
	app.AsyncReplicateMeshRequest(c.Path(), c.QueryParams(), true)
  */
};
